File: typescript/src/api/resources/judgeAlignment/client/requests/GetJudgeAlignmentJudgeAlignmentRequest.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {
 *         evaluator: "evaluator",
 *         revision: "revision"
 *     }
 */
export interface GetJudgeAlignmentJudgeAlignmentRequest {
    evaluator: string;
    revision: string;
    environment?: Sikaru.GetJudgeAlignmentJudgeAlignmentRequestEnvironment;
}

File: ruby/lib/Sikaru/judge_alignment/types/get_judge_alignment_judge_alignment_request.rb
# frozen_string_literal: true

module Sikaru
  module JudgeAlignment
    module Types
      class GetJudgeAlignmentJudgeAlignmentRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :evaluator, -> { String }, optional: false, nullable: false

        field :revision, -> { String }, optional: false, nullable: false

        field :environment, -> { Sikaru::JudgeAlignment::Types::GetJudgeAlignmentJudgeAlignmentRequestEnvironment }, optional: true, nullable: false
      end
    end
  end
end
